helm_deploy/hmpps-approved-premises-api/templates/cronjob-prod-to-preprod-refresh.yaml

```diff
@@ -0,0 +1,97 @@
+{{- if .Values.env_details.production_env -}}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: db-refresh-script
+data:
+  entrypoint.sh: |-
+    #!/bin/bash
+    set -e
+
+    echo "${DB_HOST}:5432:${DB_NAME}:${DB_USER}:${DB_PASS}" > ~/.pgpass
+    echo "${DB_HOST_PREPROD}:5432:${DB_NAME_PREPROD}:${DB_USER_PREPROD}:${DB_PASS_PREPROD}" >> ~/.pgpass
+    chmod 0600 ~/.pgpass
+
+    set -x
+    pg_dump --host="$DB_HOST" --username="$DB_USER" --format=custom --no-privileges --verbose --file=/tmp/db.dump "$DB_NAME"
+    pg_restore --host="$DB_HOST_PREPROD" --username="$DB_USER_PREPROD" --clean --no-owner --verbose --dbname="$DB_NAME_PREPROD" /tmp/db.dump
+
+    rm -v /tmp/db.dump ~/.pgpass
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: db-refresh-job
+spec:
+  schedule: "0 8 * * 0"
+  concurrencyPolicy: "Forbid"
+  jobTemplate:
+    spec:
+      backoffLimit: 0
+      activeDeadlineSeconds: 1200
+      template:
+        spec:
+          {{- if .Values.serviceAccountName }}
+          serviceAccountName: {{ .Values.serviceAccountName }}
+          {{- end }}
+          securityContext:
+            runAsUser: 999
+          containers:
+            - name: dbrefresh
+              image: "postgres:14"
```

Comment:
Is this the version of postgres we use? I think we use postgis:postgis too so I'm not sure if there are risks to using this inconsistently?

Reply:
The RDS instance we use in the Cloud Platform is vanilla Postgres, version 14. As we're only doing client actions in this container, I don't think it matters.